walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,109 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
+/**
+ * FindMeetingQuery is the "find a meeting" feature class for when given the meeting
+ * information, return the times when the meeting could happen that day.
+*/
 public final class FindMeetingQuery {
+
+  /** 
+   * Query timeslots where all attendees must be free to attend the meeting.
+   * Worst-case runtime: O(events * attendees).
+  */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+
+    // Edge case: No attendees.
+    if (attendees.isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    long meetingDuration = request.getDuration();
```

Comment:
nit: i think you can just use request.GetDuration() directly rather than creating a var for it.

Reply:
Done